### PR TITLE
Feature/149 feature editor in al sidebar

### PR DESCRIPTION
**Pull request by matchlesswei**
A user can change annotation directly in AL sidebar.
screen shot 2018-05-20 at 15 36 16

### CODE REVIEW

**Comment by betoboullosa** [comment]
@matchlesswei there are some conflicts here. Apart of that, is it ready for review?

**Comment by matchlesswei** [comment]
@betoboullosa apart of the conflicts, it is ready for review.

**Comment by betoboullosa** [comment]
@matchlesswei There is another branch with almost the same name. Can it be deleted?

https://github.com/inception-project/inception/tree/feature/149-feature-editor-directly-in-AL-sidebar

https://github.com/inception-project/inception/tree/feature/149-feature-editor-in-alSidebar

File: inception-active-learning/src/main/java/de/tudarmstadt/ukp/inception/active/learning/sidebar/ActiveLearningSidebar.java [hunk 1]
```diff
@@ -430,45 +459,49 @@ private void writeLearningRecordInDatabase(LearningRecordUserAction userAction)
...
         AnnotatorState annotatorState = ActiveLearningSidebar.this.getModelObject();
         JCas jCas = this.getJCasProvider().get();
         SpanAdapter adapter = (SpanAdapter) annotationService.getAdapter(selectedLayer.getObject());
-        AnnotationObject acceptedRecommendation = currentRecommendation;
-        int begin = acceptedRecommendation.getOffset().getBeginCharacter();
-        int end = acceptedRecommendation.getOffset().getEndCharacter();
+        int begin = currentRecommendation.getOffset().getBeginCharacter();
+        int end = currentRecommendation.getOffset().getEndCharacter();
         int id = adapter.add(annotatorState, jCas, begin, end);
+        // Get feature from recommendation
```
Comment (betoboullosa) [inline inception-active-learning/src/main/java/de/tudarmstadt/ukp/inception/active/learning/sidebar/ActiveLearningSidebar.java]: Insert an empty line before this comment to make it more readable.

**Comment by matchlesswei** [comment]
Yes, thanks @reckart , I'm now working on issue 271 adding AnnotationFeature to learning record database.

**Comment by reckart** [comment]
@matchlesswei where are we here? Can you please update the branch and resolve conflicts?

**Comment by matchlesswei** [comment]
I'm waiting for #292 (changes in the learning history database). And #292 is waiting for #294 to be merged. Because after merging, there will be new conflicts. 
Or I can merge #294 into the branch of #292. And then merge #292 into this branch.

**Pull request merged**